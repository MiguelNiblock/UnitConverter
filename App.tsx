/**
 * @format
 */

import React from 'react';
import {
  ScrollView,
  StatusBar,
  StyleSheet
} from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import Form from './src/screens/form'
import {UnitProvider} from './src/context/unitContext'

const App = () => {

  return (
    <SafeAreaProvider>
      <StatusBar barStyle={'light-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        contentContainerStyle={styles.scroll}
      >
          <UnitProvider>
            <Form/>
          </UnitProvider>
      </ScrollView>
    </SafeAreaProvider>
  );
};

const styles = StyleSheet.create({
  scroll: {
    flex:1,
  }
})

export default App;
