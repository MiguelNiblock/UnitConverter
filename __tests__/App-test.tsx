import 'react-native';
import React from 'react';
import App from '../App';
import Form from '../src/screens/form';
import Input from '../src/components/input';
import renderer from 'react-test-renderer';

describe('Component snapshots',()=>{

  it('renders App correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Form correctly', ()=>{
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  })  

  it('renders Input correctly', ()=>{
    const tree = renderer.create(<Input fieldname={"Height"} label={"ft"} amount={'20'} />).toJSON();
    expect(tree).toMatchSnapshot();
  })

})

describe('Form functionality', ()=>{

  it('gives Input the right values', ()=>{
    const tree = renderer.create(<Input fieldname={"Height"} label={"ft"} amount={'20'} />);
    expect(tree.root.props.fieldname).toBe('Height');
    expect(tree.root.props.label).toBe('ft');
    expect(tree.root.props.amount).toBe('20');    
  })
  
})