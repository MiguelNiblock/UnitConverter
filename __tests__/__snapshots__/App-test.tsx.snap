// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshots renders App correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`Component snapshots renders Form correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 40,
        "marginBottom": 40,
        "marginTop": 30,
      }
    }
  >
    Unit Converter
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "85%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Weight
      </Text>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          editable={true}
          keyboardType="decimal-pad"
          onChangeText={[Function]}
          placeholderTextColor="#86939e"
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value="0"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "fontSize": 18,
        }
      }
    >
      lbs
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "85%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#86939e",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Height
      </Text>
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          editable={true}
          keyboardType="decimal-pad"
          onChangeText={[Function]}
          placeholderTextColor="#86939e"
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="RNE__Input__text-input"
          underlineColorAndroid="transparent"
          value="0"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "fontSize": 18,
        }
      }
    >
      ft
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 40,
        "marginHorizontal": 10,
        "marginVertical": 5,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16.714285714285715,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
          >
            Imperial
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.714285714285715,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
          >
            Metric
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "alignSelf": "center",
          "margin": 25,
          "width": "45%",
        },
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component snapshots renders Input correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "85%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Height
    </Text>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        editable={true}
        keyboardType="decimal-pad"
        onChangeText={[Function]}
        placeholderTextColor="#86939e"
        style={
          Object {
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="20"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "fontSize": 18,
      }
    }
  >
    ft
  </Text>
</View>
`;
